### PR TITLE
Deprecate query, queryRowCount, and restartQuery methods

### PR DESCRIPTION
**Pull request by mattbjordan**
This will need to be backported to 3.7.x. In the backport PR, I will add a description of the changes to NextVersion.md.

@ColinKerr and I want to get this into 3.7.x, so the linter errors which say to replace the use of the now-deprecated methods have been temporarily suppressed. Here is an issue to track that the methods' usages still need to be removed/updated: #5239

### CODE REVIEW

**Review by MichaelBelousov — APPROVED** [review]
transformer changes lgtm

**Comment by aruniverse** [comment]
@mergifyio backport release/3.7.x

**Review by ColinKerr — APPROVED** [review]

File: core/backend/src/ECDb.ts [hunk 1]
```diff
@@ -311,6 +311,7 @@ export class ECDb implements IDisposable {
    * @returns Returns the query result as an *AsyncIterableIterator<any>*  which lazy load result as needed. The row format is determined by *rowFormat* parameter.
    * See [ECSQL row format]($docs/learning/ECSQLRowFormat) for details about the format of the returned rows.
    * @throws [IModelError]($common) If there was any error while submitting, preparing or stepping into query
+   * @deprecated in 3.7. Use [[createQueryReader]] instead; it accepts the same parameters.
```
Comment (pmconne) [inline core/backend/src/ECDb.ts:314]: `createQueryReader` is beta.

File: core/backend/src/IModelDb.ts [hunk 2]
```diff
@@ -518,6 +518,7 @@ export abstract class IModelDb extends IModel {
    * @returns Returns the query result as an *AsyncIterableIterator<any>*  which lazy load result as needed. The row format is determined by *rowFormat* parameter.
    * See [ECSQL row format]($docs/learning/ECSQLRowFormat) for details about the format of the returned rows.
    * @throws [IModelError]($common) If there was any error while submitting, preparing or stepping into query
+   * @deprecated in 3.7. Use [[createQueryReader]] instead; it accepts the same parameters.
```
Comment (pmconne) [inline core/backend/src/IModelDb.ts:521]: Suggested replacement API is beta.

File: core/backend/src/ECDb.ts (hunk 1 above)
Comment (mattbjordan) [reply core/backend/src/ECDb.ts:314]: Talked to Colin. We are going to bring them out of beta in 4.x.

**Review by pmconne — APPROVED** [review]

**Review by calebmshafer — APPROVED** [review]

**Review by saskliutas — APPROVED** [review]

**Pull request merged**